src/libsyntax/print/pprust.rs

```diff
@@ -2607,13 +2605,8 @@ impl<'a> State<'a> {
     }
 
     pub fn print_fn_output(&mut self, decl: &ast::FnDecl) -> IoResult<()> {
-        if let ast::Return(ref ty) = decl.output {
-            match ty.node {
-                ast::TyTup(ref tys) if tys.is_empty() => {
-                    return self.maybe_print_comment(ty.span.lo);
```

Comment:
It was intentional. As far as I can tell, this does not break anything. For example, `fn f() /*1*/ {}` still prints correctly.